manager/integration/tests/test_basic.py

```diff
@@ -3642,6 +3643,252 @@ def test_volume_toomanysnapshots_condition(client, core_api, volume_name):  # NO
+    4. Check volume data
+    5. Check if the volume still works fine by r/w data and
+       creating/removing snapshots
+    """
+    data_path1 = "/data/file1"
+    data_path2 = "/data/file2"
+
+    snap1, volume_meta_file, test_pod_name, data_md5sum1 = \
+        prepare_data_volume_metafile(client,
+                                     core_api,
+                                     volume_name,
+                                     csi_pv,
+                                     pvc,
+                                     pod,
+                                     pod_make,
+                                     data_path1)
+
+    # delete the volume metadata file of the volume on this host
+    command = ["rm", "-f", volume_meta_file]
+    subprocess.check_call(command)
+
+    # test volume functionality
+    rw_snapshot_volume_testing(client,
+                               core_api,
+                               volume_name,
+                               pod,
+                               test_pod_name,
+                               data_path1,
+                               data_md5sum1,
+                               data_path2, snap1)
+
+
+def test_volume_metafile_empty(client, core_api, volume_name, csi_pv, pvc, pod, pod_make):  # NOQA
+    """
+    Test volume metafile empty
+
+    Context:
+
+    If the volume still works fine when there is a invalid volume meta file
+    in the replica data path.
+
+    Steps:
+
+    1. Empty volume meta file in this replica data path
+    2. Recreate the pod and wait for the volume attached
+    3. Check if the volume is Healthy after the volume attached
+    4. Check volume data
+    5. Check if the volume still works fine by r/w data and
+       creating/removing snapshots
+    """
+    data_path1 = "/data/file1"
+    data_path2 = "/data/file2"
+
+    snap1, volume_meta_file, test_pod_name, data_md5sum1 = \
+        prepare_data_volume_metafile(client,
+                                     core_api,
+                                     volume_name,
+                                     csi_pv,
+                                     pvc,
+                                     pod,
+                                     pod_make,
+                                     data_path1)
+
+    # empty the volume metadata file of the volume on this host
+    command = ["truncate", "--size", "0", volume_meta_file]
+    subprocess.check_call(command)
+
+    # test volume functionality
+    rw_snapshot_volume_testing(client,
+                               core_api,
+                               volume_name,
+                               pod,
+                               test_pod_name,
+                               data_path1,
+                               data_md5sum1,
+                               data_path2, snap1)
+
+
+def test_volume_metafile_deleted_when_writing_data(client, core_api, volume_name, csi_pv, pvc, pod, pod_make):  # NOQA
```

Comment:
There seem to be duplicated codes to `prepare_data_volume_metafile`. Is it possible to add the data writing part to the  `prepare_data_volume_metafile` trigger the logic via the function argument?
```
    # create random data to the pod in the background
    command = 'dd if=/dev/urandom of=' + data_path1 + ' bs=1M' +\
              ' count=' + str(common.DATA_SIZE_IN_MB_4) + ' &'
    exec_command_in_pod(core_api, command, test_pod_name, 'default')
```

Same for the result validation part.

Reply:
Merged into the `prepare_data_volume_metafile`. Thanks.